src/audio/psp/SDL_pspaudio.c

```diff
@@ -80,10 +95,18 @@ PSPAUDIO_OpenDevice(_THIS, void *handle, const char *devname, int iscapture)
     if (this->spec.channels == 1) {
         format = PSP_AUDIO_FORMAT_MONO;
     } else {
-        this->spec.channels = 2;
         format = PSP_AUDIO_FORMAT_STEREO;
     }
-    this->hidden->channel = sceAudioChReserve(PSP_AUDIO_NEXT_CHANNEL, this->spec.samples, format);
+
+    /*  PSP has some limitations with the Audio. It fully supports 44.1KHz (Mono & Stereo),
+        however with frequencies differents than 44.1KHz, it just supports Stereo,
+        so a resampler must be done for these scenarios */
```

Comment:
You might just want to change the spec to stereo in this case, and either the application or the higher levels of SDL will do the conversion from mono to stereo if needed. SDL has optimized assembler functions for this, so that might be a performance win here.

Reply:
I'm not really sure about what you meant here.....

Let me explain better what's going on:
`PSP` natively is able to reproduce `wav` files with different frequencies than `44100Hz`, however, for these frequencies, the audio channels must be stereo.

So in order to make `mono` files work, I'm transforming mono buffer to stereo buffer, getting then better audio support.

If we remove the conversion from here, we will decrease the compatibility, then the user would need to do the transformation on his side, having the same performance impact.

Note: The transformation is just done when the **audio is mono AND with a frequency different than 44.1 kHz**, the rest of the scenarios we aren't changing anything.

Finally, you're making references to some assembly code in SDL, where is this code? and is it `MIPS ASSEMBLY`? have in mind that `PSP` is a `MIPS` architecture.

Reply:
It's not MIPS assembly, so maybe there isn't a performance benefit, but if you look at what happens in SDL_audio.c on line 1442, you'll see that it checks whether the application is allowing the driver to change the format or not, and creates a conversion pipeline if necessary. All you have to do is set the audio spec to the format actually provided by the hardware and SDL audio will do any other necessary conversion.

Reply:
As a test, try setting this->spec.channels = 2 when the application is requesting mono audio and see what happens.

Reply:
Hello @slouken ,
Thanks for the explanation and the suggestions. Sorry about that, I didn't know that SDL was offering that.

I have tried as you said to just set the
```c
this->spec.channels = 2;
``` 

However, it isn't fully working.

I have used the `loopwave.c` sample that we have under the test folder.
It is trying to reproduce a `.wav` file with channel Mono and 22 kHz.

I have changed the `SDL_OpenAudioDevice` function to allow audio changes:
```c
device = SDL_OpenAudioDevice(NULL, SDL_FALSE, &wave.spec, NULL, SDL_AUDIO_ALLOW_ANY_CHANGE);
```

The is that It sounds accelerated, I don't know if I'm doing something wrong.

I'm comparing the values to check what have changed, and as you can change just `channels` changed:
```c
obtained->freq 22050, device->spec.freq 22050
obtained->format 32784, device->spec.format 32784
obtained->channels 1, device->spec.channels 2
obtained->samples 4096, device->spec.samples 4096
```

https://user-images.githubusercontent.com/10010409/146595627-136f1134-d965-4196-aa66-53741eadc608.mov

Is SDL converting more things apart of channels?
Any clue?

Reply:
Don't set the SDL_AUDIO_ALLOW_ANY_CHANGE flag, loopwave isn't set up to handle audio format changes. However, if you set this->spec.channels = 2 and leave that flag off, SDL should create audio conversion between the application and the driver.

Reply:
So that was my first attempt, and it didn't work either.
I see how transformation is done by SDL, because I have enabled the transformation logs.

```
Converting 
AUDIO_S16
 to 
AUDIO_F32
.
Converting 
mono
 to 
stereo
.
Converting 
AUDIO_F32
 to 
AUDIO_S16
.
Converting 
AUDIO_S16
 to 
AUDIO_F32
.
Converting 
mono
 to 
stereo
.
Converting 
AUDIO_F32
 to 
AUDIO_S16
.....
```

And this is the result:
https://user-images.githubusercontent.com/10010409/146607982-14d28d9f-f5c1-4487-badd-841b83c14ca2.mov